conjur/wrapper/keystore_adapter.py

```diff
@@ -31,51 +40,63 @@ def set_password(cls, identifier, key, val):
         """
         Method for setting a password in keyring
         """
-        keyring.set_password(identifier, key, val)
+        try:
+            keyring.set_password(identifier, key, val)
+        except keyring.errors.PasswordSetError as err:
+            raise KeyringSetError(
+                f"unable to set key: {key} for identifier: {identifier}") from err
+        except keyring.errors.KeyringError as err:
+            raise KeyringGeneralError(
+                f"unable to set key: {key} for identifier: {identifier}") from err
 
     @classmethod
     def get_password(cls, identifier, key):
         """
         Method for getting a password in keyring
         """
-        return keyring.get_password(identifier, key)
+        try:
+            return keyring.get_password(identifier, key)
+        except keyring.errors.KeyringError as err:
+            raise KeyringGeneralError(
+                f"unable to get value for key: {key} for identifier: {identifier}") from err
 
-    # TODO add simple delete for one attribute only here in the adapter
     # pylint: disable=try-except-raise
     @classmethod
-    def delete_password(cls, identifier):
+    def delete_password(cls, identifier, key):
         """
         Method for deleting a password in keyring
         """
         try:
-            for attr in KEYSTORE_ATTRIBUTES:
-                keyring.delete_password(identifier, attr)
-        # TODO do not raise keyring module errors
-        # Catches when credentials do not exist in the keyring. If the key does not exist,
-        # the user has already logged out
-        except keyring.errors.PasswordDeleteError:
-            return
-        except keyring.errors.KeyringError:
-            raise
+            keyring.delete_password(identifier, key)
+        except keyring.errors.PasswordDeleteError as err:
+            raise KeyringDeletionError(
+                f"unable to delete key: {key} for identifier: {identifier}") from err
+        except keyring.errors.KeyringError as err:
+            raise KeyringGeneralError(
+                f"unable to delete key: {key} for identifier: {identifier}") from err
 
     @classmethod
     def get_keyring_name(cls):
         """
         Method to get the system's keyring name
         """
-        return keyring.get_keyring().name
+        try:
+            return keyring.get_keyring().name
+        except Exception as keyring_error:  # pylint: disable=broad-except
+            logging.debug(keyring_error)
+            return None
 
-    # TODO add a more robust check
     @classmethod
     def is_keyring_accessible(cls):
         """
         Method to check if the keyring is accessible
         """
-        # TODO investigate other errors that would make a keyring not accessible
+        # noinspection PyBroadException
```

Comment:
what is this?

Reply:
removed

Reply:
and this is a pycharm ide hint for not alerting on broad exceptions